### PR TITLE
[WIP] NIAD-132: fake spine vnp deploy

### PR DESCRIPTION
**Pull request by mattd-kainos**

### CODE REVIEW

File: integration-tests/fake_spine/fake_spine/config.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import os
+
+
+class ConfigurationException(Exception):
+    pass
+
+
+def environ_or_error(key: str, default: str = None) -> str:
+    value = os.environ.get(key, default)
+    if not value:
```
Comment (grzegorz-robak) [inline integration-tests/fake_spine/fake_spine/config.py]: You should test for None like `if value is None`, as `variable = False` will rise exception

File: integration-tests/fake_spine/fake_spine/config.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        raise ConfigurationException(f'The environment variable {key} is required.')
+    return value
+
+
+def eniron_or_error_as_int(key: str, default: str = None) -> int:
+    return int(environ_or_error(key, default))
```
Comment (grzegorz-robak) [inline integration-tests/fake_spine/fake_spine/config.py]: this seems redundant

File: integration-tests/integration_tests/README.md [hunk 3]
```diff
@@ -94,4 +94,15 @@ docker-compose -f docker-compose.yml -f docker-compose.lb.override.yml up
...
+
+## Running integration tests against fake components
+
+Run `fake_spine` and `fake_spineroutelookup` as if running component tests
+Run `inbound` and `outbound` with the `all_component_test_env.yaml` configuration (same as of running component tests)
+Run the integration tests using the `all_component_test_env.yaml` instead of the normal integration test configuration.
+
+## Fake Spine request / response delays
+
+There are two variables that can control how quickly Fake Spine responds
```
Comment (grzegorz-robak) [inline integration-tests/integration_tests/README.md]: missing description?

File: integration-tests/fake_spine/fake_spine/vnp_test_responses.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+ASYNC_EXPRESS_INBOUND_HEADERS = {
+    'Soapaction': 'urn:nhs:names:services:psisquery/QUPC_IN160102UK05'
+}
+
+
+def vnp_test_responses() -> List[MatcherAndResponses]:
+    response_mappings = []
+
+    # TODO: make the 202 response code default?
```
Comment (grzegorz-robak) [inline integration-tests/fake_spine/fake_spine/vnp_test_responses.py]: I think you did that (see SpineResponseBuilder)

**Review by grzegorz-robak — CHANGES_REQUESTED** [review]

File: integration-tests/fake_spine/fake_spine/config.py (hunk 2 above)
Comment (mattd-kainos) [reply integration-tests/fake_spine/fake_spine/config.py]: About half of the variable need conversion to into

Comment (grzegorz-robak) [reply integration-tests/fake_spine/fake_spine/config.py]: I saw that but 
`int(environ_or_error(key, default))`
`eniron_or_error_as_int(str, default)`
looks the same for me

File: integration-tests/fake_spine/fake_spine/vnp_test_responses.py (hunk 4 above)
Comment (mattd-kainos) [reply integration-tests/fake_spine/fake_spine/vnp_test_responses.py]: TODONE

**Comment by mattd-kainos** [comment]
@grzegorz-robak changed were made in https://github.com/nhsconnect/integration-adaptors/pull/183

**Comment by bartek-sarul** [comment]
moved to
https://github.com/nhsconnect/integration-adaptor-mhs/pull/5

**Pull request closed without merging**